### ISSUE TITLE
How to access introduced local variable in catch block

### ISSUE THREAD

**Issue by serkan-ozal** [jboss-javassist/javassist#361]
Hi,

I want to introduce a local variable and access it from catch block. Normally, as mentioned in the docs, it is not possible with the current API. 

But I have managed it int the following example https://github.com/serkan-ozal/javassist-issue-report/blob/master/src/main/java/ozal/serkan/javassist/issue/report/LocalVariableAccessInCatch.java 

But not sure whether it is the correct way or is there any edge cases where that approach fails.

Basically, what I do is that 
- insert local variable initialization code before
- calculate inserted code's end PC in the bytecode
- then use that PC as start PC for the exception handler

@chibash Is it safe way to achieve this?

Waiting for your feedbacks.

Regards.

--

Serkan

**Comment by JasonJunjian** [comment]
Hi, may I ask if this refactoring approach has caused any issues in actual use afterwards?
